.changesets/fix_geal_authorization_typename.md

```diff
@@ -0,0 +1,10 @@
+### special case for __typename in authorization ([PR #3821](https://github.com/apollographql/router/pull/3821))
+
+When evaluating authorization directives on fields returning interfaces,
```

Comment:
I don't fully understand this fix. Is there a ticket that goes into more detail?

Could `__typename` represent a leakage of data?

Reply:
If a concrete type is not accessible then a caller can still check to see if something exists by asking for the typename.

In the examples below a caller can see that the blog concrete type is a `PrivateBlog` even if they don't have access.

Reply:
Yes, it is expected. The authorization feature assumes that clients know about the schema, and the schema indicates the condition to access types or fields. If the data is not there and the subgraph did not return a "not found" explicitely, it's easy to infer that the type was `PrivateBlog`. If we actually want to return "not found" on unauthorized data, it is better handled at the subgraph level